tools/project_manager/editproject.php

```diff
@@ -630,26 +630,16 @@ public function save_to_db()
 
                 $details1 = 'NONE';
             }
-            $e = log_project_event($this->projectid, $GLOBALS['pguser'], 'edit', $details1);
-            if (!empty($e)) {
-                die($e);
-            }
+            $project->log_project_event($pguser, 'edit', $details1);
             if ($PPer_checkout) {
                 // we fake the project transition...
-                $e = log_project_event($this->projectid,
-                                        $GLOBALS['pguser'],
-                                        'transition',
-                                        PROJ_POST_FIRST_CHECKED_OUT,
-                                        PROJ_POST_FIRST_CHECKED_OUT,
-                                        $this->checkedoutby
-                                        );
-                if (!empty($e)) {
-                    die($e);
-                }
+                $project->log_project_event($pguser, 'transition',
+                                            PROJ_POST_FIRST_CHECKED_OUT,
+                                            PROJ_POST_FIRST_CHECKED_OUT,
+                                            $this->checkedoutby);
```

Comment:
I don't know where this is happening, but what I'm seeing is that details3 is not getting recorded in your sandbox, but it is in the main TEST sandbox:
<img width="715" alt="image" src="https://user-images.githubusercontent.com/36800239/149647171-2f8b8012-b636-48cc-8149-d3ae90a7008b.png">

The top checkout in the screenshot is from your sandbox, the bottom, a couple of minutes later, from TEST. (I checked the `project_events` table, and it's not getting recorded at all.) I double-checked using another project, and got the same results.

I've looked at ` pinc/ProjectTransition.inc`, line 299+ and at `pinc/Project.inc`, line 799+, but I'm just not seeing the problem.

Everything else I was able to test looks good!

Reply:
Can you describe your testing of this path? Was the project in "available for PP" and you edited the project and filled in the PPer?

Reply:
No, I hadn't tested that -- thanks for the idea. I just checked the projects out in the usual way: look at the list of available projects; pick one; check it out from the project page. I was doing side-by-side comparisons with checking it out via the main TEST sandbox, just to make sure I knew how it should look.

You can see the differences in the last set of events in the project history for [this](https://www.pgdp.org/c/project.php?id=projectID4aeffe8625b41) project. The checkout at `2022-01-16 04:16:08` was done from TEST, in parallel with my first checkout from your sandbox. The checkout at `2022-01-16 04:51:08` was my second checkout from your sandbox, to verify that I hadn't messed up somewhere.

Reply:
> Was the project in "available for PP" and you edited the project and filled in the PPer?

I've tested this just now to confirm what I thought was the case. Just editing the PP/PPV field (i.e. checkedoutby) in a project page will not automatically check a project out to a PPer if it is in the pool. Squirrels can then use `project_jump.php` to move it into PP checked out to that individual, but PMs and PFs cannot.

There has been some discussion of the issue in the past: editing the field in the project page is an edit operation, but moving it from available for PP to in PP checked out to someone is a project transition. Should a project transition be triggered by editing the project page?

Reply:
> Should a project transition be triggered by editing the project page?

According to line [594](https://github.com/DistributedProofreaders/dproofreaders/blob/master/tools/project_manager/editproject.php#L594) if a project was already in the PP checked out state and the PPer is changed it will check out the project to the new PPer. I just wanted to make sure I knew how you were testing it to see where I broke it. I can see the related code but I haven't figured out why my change broke it. More for me to investigate.